### PR TITLE
nextcloud28: init at 28.0.0, nextcloud27: 27.1.4 -> 27.1.5, nextcloud26: 26.0.9 -> 26.0.10

### PR DESCRIPTION
**Pull request by dotlambda**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Ma27 — CHANGES_REQUESTED** [review]
The module must be updated as well.
Can take care of that latr today.

**Comment by dotlambda** [comment]
Anything else that needs to be changed?

**Review by SuperSandro2000 — COMMENTED** [review]
Can you do https://github.com/NixOS/nixpkgs/pull/274000#discussion_r1425911424 ?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
I've deployed this to my instance and next to a new optional index I've hit the following which we should fix

```
This instance is missing some recommended PHP modules. For improved performance and better compatibility it is highly recommended to install them: bz2. For more details see the [documentation ↗](https://docs.nextcloud.com/server/28/go.php?to=admin-php-modules).
```

**Comment by SuperSandro2000** [comment]
Also I am hitting https://github.com/nextcloud/server/issues/42254

**Comment by SuperSandro2000** [comment]
Also I noticed that my desktop client started to sync external storages which crippled performance of my local lan and nextcloud. I couldn't find a bug report on this though.

**Comment by dotlambda** [comment]
> I've deployed this to my instance and next to a new optional index I've hit the following which we should fix
> 
> ```
> This instance is missing some recommended PHP modules. For improved performance and better compatibility it is highly recommended to install them: bz2. For more details see the [documentation ↗](https://docs.nextcloud.com/server/28/go.php?to=admin-php-modules).
> ```

I added all recommended PHP packages.

File: nixos/modules/services/web-apps/nextcloud.nix [hunk 1]
```diff
@@ -28,6 +28,7 @@ let
   phpPackage = cfg.phpPackage.buildEnv {
     extensions = { enabled, all }:
       (with all; enabled
+        ++ [ bz2 intl sodium ] # recommended
```
Comment (SuperSandro2000) [inline nixos/modules/services/web-apps/nextcloud.nix:31]: Small note: I've checked the php-with-extensions derivation used by nextcloud and intl and sodium where already in there but I don't think it hurts listing them twice.

**Comment by Ma27** [comment]
Also pushed the updates of nc27/nc26 here.

**Review by Ma27 — APPROVED** [review]
Just pushed a few minor changes:
* Updates to docs & release-notes
* update bookmarks app to latest patch-release since the former one was actually broken with nc28
* minor cleanups to the nextcloud module (see the commit message for a detailed explanation)

It seems to work fine for me so far.

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by Ma27** [comment]
Does @RaitoBezarius have some capacity to review this?
I'd probably wait until tomorrow (unless someone requests more time to review of course) and merge then.

**Comment by RaitoBezarius** [comment]
> Does @RaitoBezarius have some capacity to review this? I'd probably wait until tomorrow (unless someone requests more time to review of course) and merge then.

Testing now.

**Review by RaitoBezarius — APPROVED** [review]
Tested the new NC28 on a small-size production deployment (100G of data) on stable NixOS 23.11:

- Onlyoffice.
- Adding the classical missing indexes.
- The new Vue.js Files application.

Everything works, good to go for me.

**Pull request merged**